httpie/resolve.py

```diff
@@ -0,0 +1,41 @@
+import socket
+
+
+class CustomResolver:
+    def __init__(self, resolve):
+        """
+        :param resolve: A list of HOST:PORT:ADDRESS[:ADDRESS]? records
+        :type resolve: list[str]
+        """
+        self.entries = {}
+        for entry in resolve:
+            host, port, addresses = entry.split(':', 2)
```

Comment:
it seems that this won't work correctly if you specify an address without a port.  (e.g. `www.foo.com:127.0.01`) Since this address will end up as the `port` in this case.